packages/reassure-compare/src/utils/validate.ts

```diff
@@ -0,0 +1,35 @@
+import { z } from 'zod';
+import { hasDuplicateValues } from './array';
+
+const metadataSchema = z.object({
+  metadata: z.object({
+    branch: z.string(),
+    commitHash: z.string(),
+  }),
+});
+
+const performanceResultEntrySchema = z.object({
```

Comment:
Lets use naming convention that uses type name and adds `Schema` to the end.
```suggestion
const PerformanceEntrySchema = z.object({
```

Reply:
+ add `z.infer<typeof PerformanceEntrySchema>;`